File: Y2/POO/lab6/kotlin/ex3.kt


interface ILogger{
    fun logMessage(msg: String)
}

abstract class Dispozitiv(){
    var numeDispozitiv: String ="N/A"
    abstract fun esteConectat(): Boolean
}

class Imprimanta(): ILogger, Dispozitiv(){
    override fun esteConectat(): Boolean{
        return true
    }

    override fun logMessage(msg: String){
        println("Mesaj: $msg")
    }
}
fun main(){

    var imprimanta: Imprimanta = Imprimanta()

    imprimanta.logMessage("Salut")
    println(imprimanta.esteConectat())

}

File: Y2/POO/lab6/ex3.cpp
#include <iostream>
#include <string>

using namespace std;

class ILogger{
    public:
    virtual void logMessage(string msg) = 0;

    virtual ~ILogger() = default;
};

class Dispozitiv{
    protected:
        string numeDispozitiv;
    
        virtual bool esteConectat() = 0;

    public:
        string getNume() const{
            return this->numeDispozitiv;
        }
};

class Imprimanta: public ILogger, public Dispozitiv{
    public:
        bool esteConectat() {
            return 1;
        }

        void logMessage(string msg){
            cout<<"Mesaj: "<<msg<<endl;
        }
};
int main(){
    Imprimanta imprimanta;

    imprimanta.logMessage("Salut");
    cout<<imprimanta.esteConectat();
}